<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainDebuggerWindow</class>
 <widget class="QMainWindow" name="MainDebuggerWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>952</width>
    <height>730</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>HarukaMirai - Windows ARM64 Reverse Engineering Debugger</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>951</width>
      <height>681</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Debugger View</string>
     </attribute>
     <widget class="QListView" name="lstRegisters">
      <property name="geometry">
       <rect>
        <x>690</x>
        <y>20</y>
        <width>251</width>
        <height>381</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <bold>true</bold>
       </font>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>690</x>
        <y>0</y>
        <width>63</width>
        <height>20</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Registers:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>690</x>
        <y>400</y>
        <width>63</width>
        <height>20</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Stack:</string>
      </property>
     </widget>
     <widget class="QListView" name="lstStack">
      <property name="geometry">
       <rect>
        <x>690</x>
        <y>420</y>
        <width>251</width>
        <height>231</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <fontweight>DemiBold</fontweight>
       </font>
      </property>
     </widget>
     <widget class="QTabWidget" name="tabWidget_2">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>450</y>
        <width>691</width>
        <height>201</height>
       </rect>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab_15">
       <attribute name="title">
        <string>Command Console</string>
       </attribute>
       <widget class="QTextEdit" name="outCommandConsole">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>10</y>
          <width>641</width>
          <height>111</height>
         </rect>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
       <widget class="QLineEdit" name="lnCommand">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>130</y>
          <width>491</width>
          <height>31</height>
         </rect>
        </property>
        <property name="placeholderText">
         <string>Enter with a command example: !hk</string>
        </property>
       </widget>
       <widget class="QPushButton" name="btnSendCommand">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="geometry">
         <rect>
          <x>510</x>
          <y>130</y>
          <width>71</width>
          <height>31</height>
         </rect>
        </property>
        <property name="text">
         <string>Send</string>
        </property>
       </widget>
       <widget class="QPushButton" name="btnClear">
        <property name="geometry">
         <rect>
          <x>590</x>
          <y>130</y>
          <width>61</width>
          <height>31</height>
         </rect>
        </property>
        <property name="text">
         <string>Clear</string>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_9">
       <attribute name="title">
        <string>Memory Inspector 1</string>
       </attribute>
       <widget class="QHexView" name="memoryInspectorOne">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>681</width>
          <height>161</height>
         </rect>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_10">
       <attribute name="title">
        <string>Memory Inspector 2</string>
       </attribute>
       <widget class="QHexView" name="memoryInspectorTwo">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>681</width>
          <height>161</height>
         </rect>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_11">
       <attribute name="title">
        <string>Memory Inspector 3</string>
       </attribute>
       <widget class="QHexView" name="memoryInspectorThree">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>681</width>
          <height>161</height>
         </rect>
        </property>
       </widget>
      </widget>
     </widget>
     <widget class="HarukaDisasmView" name="tblDisasmVw">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>681</width>
        <height>451</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_14">
     <attribute name="title">
      <string>Debugger Automation</string>
     </attribute>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>120</y>
        <width>691</width>
        <height>321</height>
       </rect>
      </property>
      <property name="text">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;To automate Haruka Mirai DBG you have some options:&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;1 - Use VSCODE to Write Scripts and control the debugger using the LUA Lang.&lt;/p&gt;&lt;p&gt;2 - Use IDA Pro Integration to control and syncronize your IDA Things and HarukaMirai Things.&lt;/p&gt;&lt;p&gt;3 - Use Haruka Mirai with Binary Ninja.&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;THIS FEATURE IS ON DEVELOPMENT.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_7">
     <attribute name="title">
      <string>HW&amp;SFT Interrupt</string>
     </attribute>
     <widget class="QLabel" name="label_9">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>10</y>
        <width>291</width>
        <height>20</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Software and Hardware Interrupting List:</string>
      </property>
     </widget>
     <widget class="QTableView" name="tblInterrupts">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>40</y>
        <width>931</width>
        <height>571</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_8">
     <attribute name="title">
      <string>Call Stack</string>
     </attribute>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>10</y>
        <width>511</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Most Important Stack Tracing for the current program Context:</string>
      </property>
     </widget>
     <widget class="QListView" name="lstCallStack">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>40</y>
        <width>931</width>
        <height>571</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Memory View</string>
     </attribute>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>10</y>
        <width>381</width>
        <height>20</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Complete debugee program memory layout:</string>
      </property>
     </widget>
     <widget class="QTableView" name="tblMemoryView">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>40</y>
        <width>931</width>
        <height>571</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_3">
     <attribute name="title">
      <string>Modules</string>
     </attribute>
     <widget class="QListView" name="lstModules">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>40</y>
        <width>931</width>
        <height>271</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>10</y>
        <width>301</width>
        <height>20</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>All Modules for the debuggee process:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>320</y>
        <width>361</width>
        <height>20</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>All past-loaded Modules for the debugee process:</string>
      </property>
     </widget>
     <widget class="QListView" name="lstUnloadedModules">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>350</y>
        <width>931</width>
        <height>261</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_4">
     <attribute name="title">
      <string>Threads</string>
     </attribute>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>10</y>
        <width>281</width>
        <height>20</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>All Threads for the debuggee process:</string>
      </property>
     </widget>
     <widget class="QListView" name="lstThreads">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>50</y>
        <width>931</width>
        <height>561</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_5">
     <attribute name="title">
      <string>Handles</string>
     </attribute>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>10</y>
        <width>361</width>
        <height>20</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Completely Handle table for the debuggee process:</string>
      </property>
     </widget>
     <widget class="QTableView" name="tblHandles">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>40</y>
        <width>931</width>
        <height>561</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_13">
     <attribute name="title">
      <string>Process Container Callbacks</string>
     </attribute>
     <widget class="QListView" name="lstRegisteredVehs">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>490</y>
        <width>931</width>
        <height>151</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>460</y>
        <width>181</width>
        <height>20</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Process registered VEHs:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_12">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>10</y>
        <width>181</width>
        <height>20</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Process Callbacks:</string>
      </property>
     </widget>
     <widget class="QListView" name="lstProcessCallbacks">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>40</y>
        <width>931</width>
        <height>411</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_6">
     <attribute name="title">
      <string>Pdb Inpector</string>
     </attribute>
     <widget class="QPushButton" name="btnOpenPdb">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>151</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>Load PDB</string>
      </property>
     </widget>
     <widget class="QTableView" name="tblPdbFunctions">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>70</y>
        <width>931</width>
        <height>561</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="lblPdbInspectorMetrics">
      <property name="geometry">
       <rect>
        <x>730</x>
        <y>20</y>
        <width>211</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QPushButton" name="btnClearPdb">
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>10</y>
        <width>151</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>Clear Info</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_16">
     <attribute name="title">
      <string>Decompiler</string>
     </attribute>
     <widget class="QTextEdit" name="txtDecompiler">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>941</width>
        <height>641</height>
       </rect>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>952</width>
     <height>26</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <bold>true</bold>
    </font>
   </property>
   <widget class="QMenu" name="menuHarukaMirai_Debugger">
    <property name="font">
     <font>
      <bold>false</bold>
     </font>
    </property>
    <property name="title">
     <string>HarukaMirai</string>
    </property>
    <addaction name="btnOpenExecutable"/>
    <addaction name="btnAttachProcessContainer"/>
    <addaction name="separator"/>
    <addaction name="btnExit"/>
   </widget>
   <widget class="QMenu" name="menuUtils">
    <property name="title">
     <string>Utils</string>
    </property>
    <addaction name="btnMemoryPatchs"/>
    <addaction name="btnExportPatchs"/>
    <addaction name="btnImportPatchs"/>
    <addaction name="btnColorMode"/>
    <addaction name="actionARM_Processor_Features"/>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="btnAbout"/>
   </widget>
   <widget class="QMenu" name="menuDebugCommands">
    <property name="title">
     <string>Debug Commands</string>
    </property>
    <addaction name="btnRun"/>
    <addaction name="btnStepIn"/>
    <addaction name="btnStepOut"/>
    <addaction name="btnStepOver"/>
    <addaction name="btnStop"/>
   </widget>
   <addaction name="menuHarukaMirai_Debugger"/>
   <addaction name="menuDebugCommands"/>
   <addaction name="menuUtils"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="btnOpenExecutable">
   <property name="text">
    <string>Open Executable</string>
   </property>
   <property name="font">
    <font/>
   </property>
  </action>
  <action name="btnAttachProcessContainer">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Attach Process Container</string>
   </property>
  </action>
  <action name="btnDebugDynamicLibrary">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Debug a Dynamic Link Library</string>
   </property>
  </action>
  <action name="btnExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="btnMemoryPatchs">
   <property name="text">
    <string>Memory Patchs</string>
   </property>
  </action>
  <action name="btnExportPatchs">
   <property name="text">
    <string>Export Patchs</string>
   </property>
  </action>
  <action name="btnImportPatchs">
   <property name="text">
    <string>Import Patchs</string>
   </property>
  </action>
  <action name="btnAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="btnRun">
   <property name="text">
    <string>Run</string>
   </property>
  </action>
  <action name="btnStepIn">
   <property name="text">
    <string>Step Inside</string>
   </property>
  </action>
  <action name="btnStepOut">
   <property name="text">
    <string>Step Out</string>
   </property>
  </action>
  <action name="btnStepOver">
   <property name="text">
    <string>Step Over</string>
   </property>
  </action>
  <action name="btnStop">
   <property name="text">
    <string>Stop</string>
   </property>
  </action>
  <action name="btnColorMode">
   <property name="text">
    <string>Dark Mode</string>
   </property>
  </action>
  <action name="actionARM_Processor_Features">
   <property name="text">
    <string>ARM Processor Features</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>HarukaDisasmView</class>
   <extends>QTableView</extends>
   <header>debuggerwidgets\custom\disasmview\harukadisasmview.hh</header>
  </customwidget>
  <customwidget>
   <class>QHexView</class>
   <extends>QListWidget</extends>
   <header>debuggerwidgets\custom\qhexview\QHexView.hh</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
